libs/common/Networking/TcpNetworkHandlerBase.cs

```diff
@@ -135,31 +135,58 @@ void Dispose(SocketAsyncEventArgs e)
         }
 
         void RecvEventArg_Completed(object sender, SocketAsyncEventArgs e)
+        {
+            if (e.BytesTransferred == 0 || e.SocketError != SocketError.Success || serverHook.Disposed)
+            {
+                // No more things to receive
+                Dispose(e);
+                return;
+            }
+            var networkBufferProcessingTask = OnNetworkReceiveAsync(e.BytesTransferred);
+            if (networkBufferProcessingTask.IsCompletedSuccessfully)
```

Comment:
This should be .IsCompleted (i.e., it is no longer running, can be success or canceled or faulted)

Reply:
I found a way to make this work with async/await rather than continuewith and handle these conditions. So now if task throws an exception or gets cancelled (throwing an OperationCancelledException), both should get handled and success path will work as usual.